label_studio/tasks/models.py

```diff
@@ -158,6 +158,10 @@ def has_lock(self, user=None):
                     num_annotations=num_annotations,
                 )
             )
+            if self.is_labeled == False:
+                self.update_is_labeled()
+                if self.is_labeled == True:
+                    self.save(update_fields=['is_labeled'])
```

Comment:
It feels a bit counterintuitive to me that this function (named `has_lock`, and where the docstring says it just checks whether the task is locked) would update the task / potentially call `enrollment` as well. If we really need to save here, I would suggest updating the docstring explaining the rationale, and adding a unit test capturing the impact of the change.

Reply:
@jombooth

I agree, ideally `has_lock` should not do anything else. 

Added test and fixed the docstring.